src/System.Linq.Parallel/tests/ExchangeTests.cs

```diff
@@ -131,22 +132,28 @@ public static void Merge_Ordered_Longrunning(Labeled<ParallelQuery<int>> labeled
         }
 
         [Theory]
-        [MemberData("MergeData", (object)(new int[] { 16, 1024 }))]
-        [MemberData("ThrowOnCount_AllMergeOptions_MemberData", (object)(new int[] { 16, 1024 }))]
+        [MemberData("ThrowOnCount_AllMergeOptions_MemberData", (object)(new int[] { 4, 8 }))]
         // FailingMergeData has enumerables that throw errors when attempting to perform the nth enumeration.
         // This test checks whether the query runs in a pipelined or buffered fashion.
         public static void Merge_Ordered_Pipelining(Labeled<ParallelQuery<int>> labeled, int count, ParallelMergeOptions options)
         {
-            Assert.Equal(0, labeled.Item.WithMergeOptions(options).Select(x => x).First());
+            int partitions = Environment.ProcessorCount == 1 ? 1 : count - 1;
```

Comment:
I think I feel a little silly about it now.  Stemmed from some of the other tests which check against processor count for default parallelization checks, but that's not likely to be an issue here.  Potentially `count` could be adjusted based on processor count, but since we're messing with DOP anyways...